api/v1alpha1/thinruntime_types.go

```diff
@@ -0,0 +1,211 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	ThinRuntimeKind = "ThinRuntime"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// ThinRuntimeSpec defines the desired state of ThinRuntime
+type ThinRuntimeSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	ThinRuntimeProfileName string `json:"profileName,omitempty"`
+
+	// file system of thinRuntime
+	FileSystemType string `json:"fileSystemType"`
+
+	// The version information that instructs fluid to orchestrate a particular version,
+	Version VersionSpec `json:"version,omitempty"`
+
+	// The component spec of worker
+	Worker ThinCompTemplateSpec `json:"worker,omitempty"`
+
+	// The component spec of thinRuntime
+	Fuse ThinFuseSpec `json:"fuse,omitempty"`
```

Comment:
Some storage, for example NFS, does not use Fuse. Maybe it's better to use something like `DataClient` or `PosixClient` instead of `Fuse`?